cmd/cluster-init/update_jobs.go

```diff
@@ -0,0 +1,130 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/sirupsen/logrus"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	IPFilePath      = "ci-operator/jobs/infra-periodics.yaml"
+	PrePostFilePath = "ci-operator/jobs/openshift/release"
+)
+
+type InfraPeriodics struct {
+	Periodics []prowconfig.Periodic `json:"periodics,omitempty"`
+}
+
+type Post struct {
+	OSRelease struct {
+		Jobs []prowconfig.Postsubmit `json:"openshift/release,omitempty"`
+	} `json:"postsubmits,omitempty"`
+}
+
+type Pre struct {
+	OSRelease struct {
+		Jobs []prowconfig.Presubmit `json:"openshift/release,omitempty"`
+	} `json:"presubmits,omitempty"`
+}
+
+func updatePresubmits(o options) error {
```

Comment:
Look at using `jc.WriteToDir` - you should be able to simply worry about adding the jobs to the list, then the other abstractions deliver on writing it into the correct spots.

Reply:
It _sort of_ works, but it puts our periodics in the wrong place (`openshift-release-master-periodics.yaml`), and must use a different unmarhsaling style as the whitespace is changed. I could probably work around this by using the `jobconfig.WriteToFile` instead, but at that point are we really gaining anything? Thoughts?

Reply:
Hm. Technically speaking, the `openshift-release-master-periodics.yaml` is actually the correct place for periodics with `extra_refs` for `openshift/release@master`. Putting everything and the kitchen sink into `infra-periodics` is sloppy. The formatting it uses is the correct one (by definition) - what changes are you seeing?